examples/How_to_automate_tasks_with_functions.ipynb

```diff
@@ -0,0 +1,544 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "25750d3a",
+   "metadata": {},
+   "source": [
+    "# How to automate taks with functions (S3 bucket example)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5478248c",
+   "metadata": {},
+   "source": [
+    "This code demonstrates how to interact with chatGPT functions to perform tasks related to Amazon S3 buckets. The notebook covers key concepts such as running simple commands, searching for a specific file in all buckets, uploading a file to a specified bucket, and downloading a file from a specified bucket. \n",
+    "The OpenAI Chat API is used to generate natural language responses, understand user instructions, and generate appropriate function calls based on the user's input. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ed2fd156",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install openai\n",
+    "! pip install boto3\n",
+    "! pip install tenacity"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9617e95e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import openai\n",
+    "from tenacity import (\n",
+    "    retry,\n",
+    "    stop_after_attempt,\n",
+    "    wait_random_exponential,\n",
+    ") \n",
+    "import json\n",
+    "import boto3\n",
+    "import os\n",
+    "import datetime\n",
+    "from urllib.request import urlretrieve\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c8fb09f0",
+   "metadata": {},
+   "source": [
+    "## Initials"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6d5b1991",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "openai.api_key = ''\n",
```

Comment:
In general, we try to never encourage users to put their API keys in code. If it's ever accidentally committed or shared publicly, anyone can start using it to charge their account. We always want folks to use solutions like environment variables to store API keys.